packages/devtools/lib/src/timeline/timeline_service.dart

```diff
@@ -75,17 +75,30 @@ class TimelineService {
     // available in the engine.
     int uiThreadId;
     int gpuThreadId;
+
+    // Store the thread names for debugging purposes. If [uiThreadId] or
+    // [gpuThreadId] are null, we will print all the thread names we received
+    // to console.
+    final threadNames = [];
+
     for (TraceEvent event in events) {
-      // iOS - 'io.flutter.1.ui', Android - '1.ui'.
-      if (event.args['name'].contains('1.ui')) {
+      final name = event.args['name'];
+      threadNames.add(name);
+
+      // iOS - io.flutter.1.ui, Android - 1.ui, Dream (g3) - io.flutter.ui
+      if (name.contains('1.ui') || name.contains('flutter.ui')) {
```

Comment:
I believe that the `io.flutter.ui` / `io.flutter.gpu` names are the older names - the names the Flutter embedder used to use. Likely some embedders are still using them.

We could also try looking for something like `endsWith('.ui')` / `endsWith('.gpu')`; not sure if that will be too lenient a match.